File: apis/src/main/java/com/beat/apis/performance/application/dto/modify/performanceImage/PerformanceImageModifyResponse.java
package com.beat.apis.performance.application.dto.modify.performanceImage;

import com.beat.global.support.jackson.CdnImageUrl;

public record PerformanceImageModifyResponse(
	Long performanceImageId,
	@CdnImageUrl String performanceImage
) {

	public static PerformanceImageModifyResponse of(Long performanceImageId, String performanceImage) {
		return new PerformanceImageModifyResponse(performanceImageId, performanceImage);
	}
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/PerformanceImageModifyResponse.kt
package com.beat.apis.performance.api.response

import com.beat.apis.performance.application.result.PerformanceImageResult
import com.beat.global.support.jackson.CdnImageUrl

@ConsistentCopyVisibility
data class PerformanceImageModifyResponse private constructor(
    val performanceImageId: Long?, @field:CdnImageUrl val performanceImage: String?,
) {
    companion object {
        fun from(result: PerformanceImageResult): PerformanceImageModifyResponse =
            PerformanceImageModifyResponse(result.id, result.image)
    }
}
